src/lib/dapp/index.ts

```diff
@@ -180,16 +180,18 @@ export class DAppConnector {
    * @param pairingTopic - The pairing topic for the connection (optional).
    * @returns A Promise that resolves when the connection process is complete.
    */
-  public async openModal(pairingTopic?: string): Promise<void> {
-    const { uri, approval } = await this.connectURI(pairingTopic)
-    const walletConnectModal = new WalletConnectModal({
-      projectId: this.projectId,
-      chains: this.supportedChains,
-    })
-    walletConnectModal.openModal({ uri })
-    const session = await approval()
-    await this.onSessionConnected(session)
-    walletConnectModal.closeModal()
+  public async openModal(pairingTopic?: string): Promise<SessionTypes.Struct> {
+    try {
+      const { uri, approval } = await this.connectURI(pairingTopic)
+      this.walletConnectModal.openModal({ uri })
+      const session = await approval()
+      await this.onSessionConnected(session)
+      return session
+    } catch (error) {
+      throw error
+    } finally {
+      this.walletConnectModal.closeModal()
```

Comment:
Excellent catch with this finally block!

Reply:
Instead of being sarcastic and nitpicking, please be proactive and offer solutions. Also, it would be great if your comments included code solutions. Like other colleagues do.

Reply:
I'm sorry for the misunderstanding :( I was genuine with this comment. When errors were thrown previously, the modal stayed open and was unusable. This was an excellent find.